nixos/lib/make-system-tarball.nix

```diff
@@ -21,14 +21,33 @@
 
   # Extra tar arguments
 , extraArgs ? ""
+  # Method of compression
+, compression ? "pixz"
+  # flags passed to the compressor
+, compressorFlags ? ""
 }:
 
-stdenv.mkDerivation {
+let
+  extensionMap = {
+    xz = ".xz";
+    pixz = ".xz";
+    gzip = ".gz";
+    pigz = ".gz";
+    bzip2 = ".bz2";
+    lzma = ".lzma";
+    none = "";
+  };
+  inputMap = {
+    inherit xz pixz gzip pigz bzip2 lzma;
+    none = null;
+  };
```

Comment:
I don't love that this isn't extendable by the caller, but perhaps it is not necessary?

Reply:
i could also just add some more arguments, that default to looking things up in the map, so you can bypass the map easily?